pyatlan/model/typedef.py

```diff
@@ -717,6 +781,8 @@ def create(
         applicable_glossaries: Optional[Set[str]] = None,
         applicable_glossary_types: Optional[Set[str]] = None,
         applicable_other_asset_types: Optional[Set[str]] = None,
+        applicable_domains: Optional[Set[str]] = None,
+        applicable_domain_types: Optional[Set[str]] = None,
```

Comment:
I think the type should be Optional[Set[Literal["DataDomain", "DataProduct"]]] not Optional[Set[str]]